vllm/v1/worker/gpu_model_runner.py

```diff
@@ -788,16 +795,24 @@ def _prepare_inputs(
 
         # Prepare encoder attention metadata separately
         # (encoder layers are not in KV cache groups)
-        if self.is_encoder_only_model:
-            common_attn_metadata, encoder_attn_metadata = \
-                self._build_encoder_only_attn_metadata(
-                scheduler_output)
+        if self.is_encoder_only_model or (
+                self.model_config.is_encoder_decoder
+                and scheduler_output.scheduled_encoder_inputs):
+            if self.is_encoder_only_model:
+                common_attn_metadata, encoder_attn_metadata = \
+                    self._build_encoder_only_attn_metadata(
+                    scheduler_output)
+            else:
+                common_attn_metadata, encoder_attn_metadata = \
+                    self._build_enc_dec_attn_metadata(
+                    scheduler_output)
```

Comment:
I think we can simplify to

```suggestion
        if self.is_encoder_only_model:
            common_attn_metadata, encoder_attn_metadata = \
                    self._build_encoder_only_attn_metadata(
                    scheduler_output)
        elif self.model_config.is_encoder_decoder and scheduler_output.scheduled_encoder_inputs:
             common_attn_metadata, encoder_attn_metadata = \
                    self._build_enc_dec_attn_metadata(
                    scheduler_output)
```